utils/issues/issuescollection.go

```diff
@@ -0,0 +1,362 @@
+	}
+	if len(issues.SecretsViolations) > 0 {
+		ic.SecretsViolations = append(ic.SecretsViolations, issues.SecretsViolations...)
+	}
+	// Sast
+	if len(issues.SastVulnerabilities) > 0 {
+		ic.SastVulnerabilities = append(ic.SastVulnerabilities, issues.SastVulnerabilities...)
+	}
+	if len(issues.SastViolations) > 0 {
+		ic.SastViolations = append(ic.SastViolations, issues.SastViolations...)
+	}
+	// Iac
+	if len(issues.IacVulnerabilities) > 0 {
+		ic.IacVulnerabilities = append(ic.IacVulnerabilities, issues.IacVulnerabilities...)
+	}
+	if len(issues.IacViolations) > 0 {
+		ic.IacViolations = append(ic.IacViolations, issues.IacViolations...)
+	}
+}
+
+func (ic *ScansIssuesCollection) AppendStatus(scanStatus formats.ScanStatus) {
+	if ic.ScaStatusCode == nil || (*ic.ScaStatusCode == 0 && scanStatus.ScaStatusCode != nil) {
+		ic.ScaStatusCode = scanStatus.ScaStatusCode
+	}
+	if ic.IacStatusCode == nil || (*ic.IacStatusCode == 0 && scanStatus.IacStatusCode != nil) {
+		ic.IacStatusCode = scanStatus.IacStatusCode
+	}
+	if ic.SecretsStatusCode == nil || (*ic.SecretsStatusCode == 0 && scanStatus.SecretsStatusCode != nil) {
+		ic.SecretsStatusCode = scanStatus.SecretsStatusCode
+	}
+	if ic.SastStatusCode == nil || (*ic.SastStatusCode == 0 && scanStatus.SastStatusCode != nil) {
+		ic.SastStatusCode = scanStatus.SastStatusCode
+	}
+	if ic.ApplicabilityStatusCode == nil || (*ic.ApplicabilityStatusCode == 0 && scanStatus.ApplicabilityStatusCode != nil) {
+		ic.ApplicabilityStatusCode = scanStatus.ApplicabilityStatusCode
+	}
+}
+
+func (ic *ScansIssuesCollection) IsScanNotCompleted(scanType utils.SubScanType) bool {
+	status := ic.GetScanStatus(scanType)
+	// Failed or not performed scans
+	return status == nil || *status != 0
+}
+
+func (ic *ScansIssuesCollection) GetScanStatus(scanType utils.SubScanType) *int {
+	switch scanType {
+	case utils.ScaScan:
+		return ic.ScaStatusCode
+	case utils.IacScan:
+		return ic.IacStatusCode
+	case utils.SecretsScan:
+		return ic.SecretsStatusCode
+	case utils.SastScan:
+		return ic.SastStatusCode
+	case utils.ContextualAnalysisScan:
+		return ic.ApplicabilityStatusCode
+	}
+	return nil
+}
+
+// Only if performed and failed
+func (ic *ScansIssuesCollection) HasErrors() bool {
+	if scaStatus := ic.GetScanStatus(utils.ScaScan); scaStatus != nil && *scaStatus != 0 {
+		return true
+	}
+	if applicabilityStatus := ic.GetScanStatus(utils.ContextualAnalysisScan); applicabilityStatus != nil && *applicabilityStatus != 0 {
+		return true
+	}
+	if iacStatus := ic.GetScanStatus(utils.IacScan); iacStatus != nil && *iacStatus != 0 {
+		return true
+	}
+	if secretsStatus := ic.GetScanStatus(utils.SecretsScan); secretsStatus != nil && *secretsStatus != 0 {
+		return true
+	}
+	if sastStatus := ic.GetScanStatus(utils.SastScan); sastStatus != nil && *sastStatus != 0 {
+		return true
+	}
+	return false
+}
+
+func (ic *ScansIssuesCollection) GetScanDetails(scanType utils.SubScanType, violation bool) map[severityutils.Severity]int {
```

Comment:
```suggestion
func (ic *ScansIssuesCollection) GetScanDetails(scanType utils.SubScanType, isViolation bool) map[severityutils.Severity]int {
```

change name to avoid inner name collision